driver/ppm_fillers.c

```diff
@@ -4501,3 +4510,163 @@ static int f_sys_signaldeliver_e(struct event_filler_arguments *args)
 	return add_sentinel(args);
 }
 #endif
+
+static inline u16 semop_flags_to_scap(short flags)
+{
+	u16 res = 0;
+
+	if (flags & IPC_NOWAIT)
+		res |= PPM_IPC_NOWAIT;
+
+	if (flags & SEM_UNDO)
+		res |= PPM_SEM_UNDO;
+
+	return res;
+}
+
+static int f_sys_semop_e(struct event_filler_arguments *args)
+{
+	unsigned long val;
+	int res;
+        struct sembuf *ptr;
+
+	/*
+	 * semid
+	 */
+	syscall_get_arguments(current, args->regs, 0, 1, &val);
+	res = val_to_ring(args, val, 0, true, 0);
+	if (unlikely(res != PPM_SUCCESS))
+		return res;
+
+	/*
+	 * sembuf
+	 */
+	syscall_get_arguments(current, args->regs, 1, 1, (unsigned long*) &ptr);
```

Comment:
In this case you are getting a raw pointer from userspace memory and using as is. It may cause kernel panic if it generates a pagefault. Please use `ppm_copy_from_user` like we did [here](https://github.com/draios/sysdig/blob/dev/driver/ppm_fillers.c#L4387). Also in this case, as I have said on a previous comment, we put this data on the syscall exit event, because it's more probable that the access does not generate page faults.